### ISSUE TITLE
JForm::findField rename $element loop variable

### ISSUE THREAD

**Issue by dr138** [joomla/joomla-cms#9435]
#### Steps to reproduce the issue
The
```php
foreach ($elements as $element)
{
...
}
```
loop within findField should have the loop variable renamed to (eg) $el, otherwise it can end up being returned as the function return value.

Eg for
```php
$jform = new JForm('testform');
$jform->load(<<<'XML'
<form>
  <fields name="A">
    <field name="fielda"/>
    <fields name="B">
      <field name="fieldab"/>
    </fields>
  </fields>
</form>
XML
);
$field = $jform->findField('fieldab','A'); // should return false, but returns whole <fields name="A"> element.
print ($field? $field->asXML()."\n": "false")."\n";
```

And then for
```php
$jform->load(<<<'XML'
<form>
<fields name="A">
  <field name="fieldab"/>
</fields>
</form>
XML
,true); // should add a new <field name="fieldab"/> element to the <fields name="A"> element, instead deletes the whole original <fields name="A"> element and adds the <field> at the "top" level.
print $jform->getxml()->asXML();
```
David
#### Expected result
See commented code above

#### Actual result
See commented code above

**Comment by izharaazmi** [comment]
I confirm this issue exists.

**But....**, the given code won't execute from outside `JForm` as `findField` is a protected method. Using `getField` will still return false because field setup fails at `libraries/joomla/form/form.php:1823` as `libraries/joomla/form/field.php:557` returns false.

@dr138 Please send a PR.

**Comment by brianteeman** [comment]
Closed as we have a PR #11732
